### ISSUE TITLE
Let's talk design!

### ISSUE THREAD

**Issue by fasterthanlime** [itchio/capsule#1]
Here are the requirements:
- Should be able to display small overlay in a corner of the screen to show recording controls
- Should be relatively light: dropping a few FPS, not making a game unplayable (very vague req., will have to see how real-life code performs)
- Should be usable when target is started before capsule or after capsule - ie. capsule should be able to attach/detach itself from processes (and detect if they're using OpenGL/D3D/Vulkan)
- Capsule doesn't need to handle transcoding itself, as long as it has a well-defined output format - maybe doing downscaling would be a good feature for capsule to allow running it on low-end systems without destroying the game's fluidity

The closest I could find to my requirements were Taksi, an open-source alternative to Fraps. I imported it to a [Github repo](https://github.com/fasterthanlime/taksi), but:
- It's super old (targets GDI/OpenGL/DX8/DX9)
- It seems broken atm - my tests with an OpenGL game captured
- It relies on windows-specific video compression codepaths
- It has a pretty bad Win32 GUI, but it sounds easy enough to strip out

Another option would be to rely on libobs (from OBS Studio), at first glance:
- It's a big codebase & already has stuff like transcoding + streaming baked in
- Uses CMake, doesn't (easily) build with mingw-w64 on Windows, relies on MSVC still (my first attempts at compiling it on Windows were unsuccessful)
- It's designed with cross-platform support in mind
- It has no need/desire/support for overlays

TL;DR there seems to be two main strategies:
- bring taksi up to speed / migrate it to a sane build system or just do a fresh project based on some of the same tricks
- trim down libobs until it's small enough to fit the reqs + add overlay support to it

**Comment by fasterthanlime** [comment]
Current status: using neither taksi nor libobs, but taking some inspiration from libobs.

**Comment by roxlu** [comment]
@fasterthanlime are you on IRC? Would love to share some thoughts.

**Comment by fasterthanlime** [comment]
@roxlu I'm "fasterthanlime" on freenode :)
